src/Microsoft.DocAsCode.Build.Common/Reference/OverwriteDocumentModel.cs

```diff
@@ -57,6 +58,20 @@ public class OverwriteDocumentModel
         [JsonIgnore]
         public HashSet<string> LinkToUids { get; set; } = new HashSet<string>();
 
+        /// <summary>
+        /// Link sources information for file
+        /// </summary>
+        [YamlIgnore]
+        [JsonIgnore]
+        public ImmutableDictionary<string, ImmutableList<LinkSourceInfo>> FileLinkSources { get; set; } = ImmutableDictionary<string, ImmutableList<LinkSourceInfo>>.Empty;
```

Comment:
Immutable type cannot serialize in binary format, it is depended by 32-bit process.